modules/core/src/camera/vpColorDepthConversion.cpp

```diff
@@ -0,0 +1,194 @@
+/****************************************************************************
+ *
+ * ViSP, open source Visual Servoing Platform software.
+ * Copyright (C) 2005 - 2019 by Inria. All rights reserved.
+ *
+ * This software is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ * See the file LICENSE.txt at the root directory of this source
+ * distribution for additional information about the GNU GPL.
+ *
+ * For using ViSP with software that can not be combined with the GNU
+ * GPL, please contact Inria about acquiring a ViSP Professional
+ * Edition License.
+ *
+ * See http://visp.inria.fr for more information.
+ *
+ * This software was developed at:
+ * Inria Rennes - Bretagne Atlantique
+ * Campus Universitaire de Beaulieu
+ * 35042 Rennes Cedex
+ * France
+ *
+ * If you have questions regarding the use of this file, please contact
+ * Inria at visp@inria.fr
+ *
+ * This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+ * WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * Description:
+ * Color to depth conversion.
+ *
+ * Authors:
+ *
+ *****************************************************************************/
+
+/*!
+  \file vpColorDepthConversion.cpp
+  \brief color to depth conversion
+*/
+
+#include <visp3/core/vpColorDepthConversion.h>
+
+// System
+#include <algorithm>
+
+// Internal
+#include <visp3/core/vpMeterPixelConversion.h>
+#include <visp3/core/vpPixelMeterConversion.h>
+
+namespace
+{
+
+/*!
+ * Adjust 2D image point to the image boundary.
+ *
+ * \param[in] ip : 2D point to adjust.
+ * \param[in] width : Image width.
+ * \param[in] height : Image height.
+ * \return Adjusted 2D point (<=> [i,j] in [0,height][0,width] ).
+ */
+vpImagePoint adjust2DPointToBoundary(const vpImagePoint &ip, double width, double height)
+{
+  return {std::clamp(ip.get_i(), 0., height), std::clamp(ip.get_j(), 0., width)};
+}
```

Comment:
Use rather `vpMath::clamp()` that calls internally  `std::clamp()` if c++17 available, otherwise use default code 

Ca donnerait quelque chose du type (pas testé):
```
/*!
Clamp a value to boundaries.
\param v : The value to clamp.
\param lower, upper : The boundaries to clamp `v` to.

The behavior is undefined if the value of `lower` is greater than `upper`. 

template <typename T>
static T vpMath::clamp(const T& v, const T& lower, const T& upper) {
#if (VISP_CXX_STANDARD >= VISP_CXX_STANDARD_17)
  return std::clamp(v, lower, upper);
#else
  return ((std::max))(lower, ((std::min))(v, upper));
#endif
}
```
Les doubles parenthèses autour de `std::max` et `std::min` sont là exprès pour des question de compatibilité avec des vieilles version du compilateur (si je me souviens bien).

Reply:
Done avec un code de test dans `visp/modules/core/test/math/testMath.cpp`